docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,369 @@
+
+## Creating, Deleting, and Updating Seeds Declaratively
+
+When all or most of the existing seeds are near capacity, new seeds should be created in order to accommodate more shoots. Conversely, sometimes there could be too many seeds for the number of shoots, and so some of the seeds could be deleted to save resources. Currently, the process of creating a new seed involves a number of manual steps, such as creating a new shoot that meets certain criteria, and then registering it as a seed in Gardener. This could be automated to some extent by [annotating a shoot with the `use-as-seed` annotation](../usage/shooted_seed.md), in order to create a "shooted seed". However, adding more than one similar seeds still requires manually creating all needed shoots, annotating them appropriately, and making sure that they are successfully reconciled and registered.
+
+To create, delete, and update seeds effectively in a declarative way and allow auto-scaling, a "creatable seed" resource along with a "set" (and in the future, perhaps also a "deployment") of such creatable seeds should be introduced, similar to Kubernetes `Pod`, `ReplicaSet`, and `Deployment` (or to MCM `Machine`, `MachineSet`, and `MachineDeployment`) resources. With such resources (and their respective controllers), creating a new seed based on a template would become as simple as increasing the `replicas` field in the "set" resource. 
+
+In [Issue #2181](https://github.com/gardener/gardener/issues/2181) it is already proposed that the `use-as-seed` annotation is replaced by a dedicated `ShootedSeed` resource. The solution proposed here further elaborates on this idea.
+
+### ShootedSeeds
+
+#### ShootedSeed Resource
+
+The `ShootedSeed` resource is a dedicated custom resource that represents a "shooted seed" and properly replaces the `use-as-seed` annotation. This resource contains:
+
+* A `shoot` section that contains the Shoot spec and parts of the metadata, such as labels and annotations.
+* An optional `seed` section that contains the Seed spec and parts of the metadata, such as labels and annotations.
+* An optional `gardenlet` section that contains:
+    * Certain aspects of the `gardenlet` deployment configuration, such as the number of replicas, the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+    * The `GardenletConfiguration` resource that contains controllers configuration, feature gates, etc.
+    
+Either the `seed` or the `gardenlet` section must be specified, but not both: 
+
+* If the `seed` section is specified, `gardenlet` should not be deployed in the shoot, and a new seed should be registered based on the `seed` section.
+* If the `gardenlet` section is specified, `gardenlet` should be deployed in the shoot, and it will register a new seed upon startup based on the `seedConfig` section of the `GardenletConfiguration` resource
```

Comment:
Yes, it does, and it's already the case in the proposal:

> * A `shoot` section that contains the Shoot spec and parts of the metadata, such as labels and annotations.
> * An optional `seed` section that contains the Seed spec and parts of the metadata, such as labels and annotations.

So Shoot and Seed labels and annotations are both taken from their respective sections. If `gardenlet` is specified instead of `seed`, then the annotations of the `gardenlet.config.seedConfig` are applied to the seed.

Reply:
OK, yes, thanks. It's better to have this clear separation (either `seed{Template}` or `gardenlet`) to make the API more expressive.